lightningd/peer_htlcs.c

```diff
@@ -216,12 +224,19 @@ static bool check_cltv(struct htlc_in *hin,
 	return false;
 }
 
-static void fulfill_htlc(struct htlc_in *hin, const struct preimage *preimage)
+void fulfill_htlc(struct htlc_in *hin, const struct preimage *preimage)
 {
 	u8 *msg;
 	struct channel *channel = hin->key.channel;
 	struct wallet *wallet = channel->peer->ld->wallet;
 
+	if (hin->hstate != RCVD_ADD_ACK_REVOCATION) {
```

Comment:
I assume this may happen due to the updates no longer being atomic?

Reply:
Yes, in particular we might have gone onchain while waiting, then timed out the HTLC.